SpatialGDK/Source/SpatialGDKEditorCommandlet/Private/Commandlets/CookAndGenerateSchemaCommandlet.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Commandlets/CookCommandlet.h"
+#include "CookAndGenerateSchemaCommandlet.generated.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogCookAndGenerateSchemaCommandlet, Log, All);
+
+struct FObjectListener;
+/**
+ * This Commandlet generates schema and performs a cook command.
+ * It supports the same set of arguments as cook. It will only generate
+ * schema for blueprints required by the cook.
+ */
+UCLASS()
+class SPATIALGDKEDITORCOMMANDLET_API UCookAndGenerateSchemaCommandlet : public UCookCommandlet
+{
+	GENERATED_BODY()
+
+	UCookAndGenerateSchemaCommandlet();
+
+	virtual ~UCookAndGenerateSchemaCommandlet();
+
+private:
+	FObjectListener* ObjectListener;
+
+public:
+	virtual int32 Main(const FString& CmdLineParams) override;
```

Comment:
Thinking about tests for UCookAndGenerateSchemaCommandlet I've got an idea. Can we split Main function into 2 functions: one that calls Super::Main and collects ReferencedClasses and calls the second functions, which in turn gets ReferencedClasses as an arguments and generates schema? That way we'll be able to call the latter function in Unit tests and check it's results.